subsystem/src/main/java/org/wildfly/extension/elytron/tls/subsystem/SSLContextDefinitions.java

```diff
@@ -0,0 +1,2031 @@
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition PREFER_CRLS = new SimpleAttributeDefinitionBuilder(Constants.PREFER_CRLS, ModelType.BOOLEAN, true)
+            .setDefaultValue(ModelNode.FALSE)
+            .setRequired(false)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition SOFT_FAIL = new SimpleAttributeDefinitionBuilder(Constants.SOFT_FAIL, ModelType.BOOLEAN, true)
+            .setRequired(false)
+            .setDefaultValue(ModelNode.FALSE)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition ONLY_LEAF_CERT = new SimpleAttributeDefinitionBuilder(Constants.ONLY_LEAF_CERT, ModelType.BOOLEAN, true)
+            .setRequired(false)
+            .setDefaultValue(ModelNode.FALSE)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition RESPONDER_CERTIFICATE = new SimpleAttributeDefinitionBuilder(Constants.RESPONDER_CERTIFICATE, ModelType.STRING, true)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final ObjectTypeAttributeDefinition OCSP = new ObjectTypeAttributeDefinition.Builder(Constants.OCSP, RESPONDER, PREFER_CRLS,
+            RESPONDER_CERTIFICATE, RESPONDER_KEYSTORE/* , RESPONDER_KEYSTORE_OBJECT */)
+            .setRequired(false)
+            .setRestartAllServices()
+            .build();
+
+    /** KeyManager definitions */
+
+    static final SimpleAttributeDefinition KEY_MANAGER = new SimpleAttributeDefinitionBuilder(Constants.KEY_MANAGER, ModelType.STRING, true)
+            .setMinSize(1)
+            // .setAlternatives(Constants.KEY_MANAGER_OBJECT)
+            .setCapabilityReference(KEY_MANAGER_CAPABILITY, SSL_CONTEXT_CAPABILITY)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+    static final ObjectTypeAttributeDefinition KEY_MANAGER_OBJECT = new ObjectTypeAttributeDefinition.Builder(Constants.KEY_MANAGER_OBJECT,
+    ALGORITHM, providersKMDefinition, PROVIDER_NAME, keystoreKMDefinition, keystoreKMObjectDefinition, ALIAS_FILTER, CREDENTIAL_REFERENCE,
+    GENERATE_SELF_SIGNED_CERTIFICATE_HOST)
+            .setRequired(true)
+            .setAlternatives(Constants.KEY_MANAGER)
+            .setCapabilityReference(KEY_MANAGER_CAPABILITY, SSL_CONTEXT_CAPABILITY)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+
+    /** TrustManager definitions */
+
+    static final SimpleAttributeDefinition TRUST_MANAGER = new SimpleAttributeDefinitionBuilder(Constants.TRUST_MANAGER, ModelType.STRING, true)
+            .setMinSize(1)
+            // .setAlternatives(Constants.TRUST_MANAGER_OBJECT)
```

Comment:
Same here?

Reply:
Or maybe this is something that will be added in a bit?

Reply:
same as [this comment](https://github.com/wildfly-security-incubator/elytron-tls/pull/3#discussion_r1004963789), to be added